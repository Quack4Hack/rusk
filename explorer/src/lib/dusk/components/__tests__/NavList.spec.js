import { afterEach, describe, expect, it } from "vitest";
import { cleanup, render } from "@testing-library/svelte";
import { NavList } from "..";

describe("NavList", () => {
  const baseProps = {
    navigation: [
      {
        link: "https://dusk.network",
        title: "Dusk",
      },
      {
        link: "https://explorer.dusk.network",
        title: "Explorer",
      },
    ],
  };

  afterEach(cleanup);

  it("renders the NavList component", () => {
    const { container } = render(NavList, baseProps);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render the Nav List with the correct information", () => {
    const { container } = render(NavList, baseProps);

    baseProps.navigation.forEach((item, index) => {
      expect(
        container.getElementsByClassName("dusk-nav-list__link")[index].innerHTML
      ).toBe(item.title);

      expect(
        container.getElementsByClassName("dusk-nav-list__link")[index]
      ).toHaveAttribute("href", item.link);
    });
  });

  it("should pass additional class names and attributes to the rendered element", () => {
    const props = {
      ...baseProps,
      className: "foo bar",
    };
    const { container } = render(NavList, props);

    expect(container.firstChild).toMatchSnapshot();
  });
});
