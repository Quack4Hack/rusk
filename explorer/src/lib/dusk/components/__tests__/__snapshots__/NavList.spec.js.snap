// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavList > renders the NavList component 1`] = `
<div>
  <nav
    class="dusk-nav-list"
  >
    
    <a
      class="dusk-anchor dusk-anchor--off-surface dusk-nav-list__link"
      href="https://dusk.network"
    >
      Dusk
    </a>
    <!--&lt;Anchor&gt;-->
    <!--&lt;AppAnchor&gt;-->
    
    <a
      class="dusk-anchor dusk-anchor--off-surface dusk-nav-list__link"
      href="https://explorer.dusk.network"
    >
      Explorer
    </a>
    <!--&lt;Anchor&gt;-->
    <!--&lt;AppAnchor&gt;-->
  </nav>
  
  <!--&lt;NavList&gt;-->
</div>
`;

exports[`NavList > should pass additional class names and attributes to the rendered element 1`] = `
<div>
  <nav
    class="dusk-nav-list foo bar"
  >
    
    <a
      class="dusk-anchor dusk-anchor--off-surface dusk-nav-list__link"
      href="https://dusk.network"
    >
      Dusk
    </a>
    <!--&lt;Anchor&gt;-->
    <!--&lt;AppAnchor&gt;-->
    
    <a
      class="dusk-anchor dusk-anchor--off-surface dusk-nav-list__link"
      href="https://explorer.dusk.network"
    >
      Explorer
    </a>
    <!--&lt;Anchor&gt;-->
    <!--&lt;AppAnchor&gt;-->
  </nav>
  
  <!--&lt;NavList&gt;-->
</div>
`;
